src/axom/inlet/Inlet.cpp

```diff
@@ -184,10 +190,77 @@ void Inlet::registerDocWriter(std::shared_ptr<DocWriter> writer) {
 }
 
 void Inlet::writeDoc() {
-  if (m_docWriterEnabled) {
+  if (m_docEnabled) {
     m_docWriter->writeDocumentation();
   }
 }
 
+bool Inlet::verify() {
+  bool verifySuccess = true;
+  verifyRecursive(m_sidreRootGroup, verifySuccess);
+  return verifySuccess;
+}
+
+void Inlet::verifyRecursive(axom::sidre::Group* sidreGroup, bool& verifySuccess) {
+  SLIC_ASSERT_MSG(sidreGroup, "Root was nullptr");
+  if (sidreGroup == m_sidreRootGroup && sidreGroup->hasView("warningFlag")) {
+    verifySuccess = false;
+  }
+
+  if (sidreGroup->hasView("required")) {
+    int8 required = sidreGroup->getView("required")->getData();
+    if (required && !sidreGroup->hasView("value")) {
+      std::string msg = fmt::format("Inlet: {0}: Required field was not specified in Input Deck", 
+                                    sidreGroup->getPathName());
+      SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+      verifySuccess = false;
+    }
+  }
+
+  if (sidreGroup->hasView("value")) {
+    if (sidreGroup->hasView("range")) {
+      auto type = sidreGroup->getView("range")->getTypeID();
+      if (type == axom::sidre::INT_ID) {
+        int* range = sidreGroup->getView("range")->getArray();
+        int val = sidreGroup->getView("value")->getScalar();
+        // Checks if the value is not within the specified range
+        if (!(range[0] <= val && val <= range[1])) {
+          verifySuccess = false;
+        }
+      } else if (type == axom::sidre::DOUBLE_ID) {
+        double* range = sidreGroup->getView("range")->getArray();
+        double val = sidreGroup->getView("value")->getScalar();
+        if (!(range[0] <= val && val <= range[1])) {
+          verifySuccess = false;
+        }
+      } else {
+        verifySuccess = false;
+      }
+    } else if (sidreGroup->hasView("validValues")) {
+      int val = sidreGroup->getView("value")->getScalar();
+      int* range = sidreGroup->getView("validValues")->getArray();
```

Comment:
Note: You can use ``sidre::Array`` instead.
See: https://github.com/LLNL/axom/blob/c3a64b372d25e53976b3ba8676a25acc49a9a6cd/src/axom/sidre/core/Array.hpp#L93

I think this would look something like:
```cpp
sidre::Array<int> vals = sidreGroup->getView("validValues");
for(axom::IndexType i=0; i< vals.size(); ++i)
{
   if (vals[i] == val) {
          found = true;
          break;
}
```